source/RoslynAnalyzers/Testing/Integration/IntegrationTestForwardCancellationTokenToInvocationsFixer.cs

```diff
@@ -0,0 +1,145 @@
+            {
+                return;
+            }
+
+            Task<Document> CreateChangedDocument(CancellationToken _)
+            {
+                var newRoot = UpdatedRootWithCancellationTokenSyntax(
+                    contextDocument,
+                    contextRootNode,
+                    invocationOperation,
+                    invocationSyntaxExpression,
+                    invocationSyntaxArguments
+                );
+                var newDocument = contextDocument.WithSyntaxRoot(newRoot);
+
+                return Task.FromResult(newDocument);
+            }
+
+            var title = Descriptors.Oct2008IntegrationTestForwardCancellationTokenToInvocations().Title.ToString();
+            context.RegisterCodeFix(CodeAction.Create(title, CreateChangedDocument, equivalenceKey: title), context.Diagnostics);
+        }
+
+        static async Task<IInvocationOperation?> GetInvocationOperation(Document document, SyntaxNode rootNode, TextSpan span, CancellationToken cancellationToken)
+        {
+            IInvocationOperation? invocationOperation = null;
+            var offendingNode = rootNode.FindNode(span, getInnermostNodeForTie: true);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (semanticModel == null)
+            {
+                return invocationOperation;
+            }
+
+            // The analyzer created the diagnostic on the IdentifierNameSyntax, and the parent is the actual invocation
+            if (offendingNode.Parent != null)
+            {
+                SyntaxNode? nodeToRetrieve = null;
+
+                // When method is invoked using nullability, .?Method(), then the node.Parent.Parent is the actual invocation.
+                if (offendingNode.Parent.IsKind(SyntaxKind.MemberBindingExpression))
+                {
+                    nodeToRetrieve = offendingNode.Parent.Parent;
+                }
+
+                // Use Parent.Parent for MemberBindingExpression unless if it is null.
+                invocationOperation = semanticModel.GetOperation(nodeToRetrieve ?? offendingNode.Parent, cancellationToken) as IInvocationOperation;
+            }
+
+            return invocationOperation;
+        }
+
+        static (SyntaxNode? syntaxExpression, ImmutableArray<ArgumentSyntax> syntaxArguments) GetExpressionAndArguments(SyntaxNode invocationNode)
+        {
+            SyntaxNode? syntaxExpression = null;
+            var syntaxArguments = ImmutableArray<ArgumentSyntax>.Empty;
+
+            if (invocationNode is InvocationExpressionSyntax invocationExpression)
+            {
+                syntaxExpression = invocationExpression.Expression;
+                syntaxArguments = invocationExpression.ArgumentList.Arguments.ToImmutableArray();
+            }
+
+            return (syntaxExpression, syntaxArguments);
+        }
+
+        static SyntaxNode UpdatedRootWithCancellationTokenSyntax(
```

Comment:
This has been simplified substantially when comparing to original `CA2016` implementation with many edge cases handling removed. This is because `IntegrationTest` can rely on `CA2016` + `OCT2008`